src/services/ftp/dir_stream.rs

```diff
@@ -93,7 +93,11 @@ impl futures::Stream for DirStream {
                 let d = if de.is_file() {
                     DirEntry::new(self.backend.clone(), ObjectMode::FILE, de.name())
                 } else if de.is_directory() {
-                    DirEntry::new(self.backend.clone(), ObjectMode::DIR, de.name())
+                    let mut dir_path = de.name().to_string();
```

Comment:
Seems we are going a loop here. Where returning the `DirEntry`, we should construct the correct relative name.

For example, we are listing a dir `abc/`, and we got a new dir entry called `def/`. We should return `abc/def/` instead of `def/`.

Reply:
Thanks for help! I am also aware of this, but currently I am modifying create() function to make it acts correctly. I will make this change after correction is done.